src/pages/index.js

```diff
@@ -59,25 +145,83 @@ const newCardPopup = new PopupWithForm(".popup-card", (formValues) => {
     link: formValues["second-input"],
     alt: formValues["first-input"],
   };
-  const newCard = createCard(cardData);
-
-  section.addItem(newCard, "prepend");
-  newCardPopup.close();
-  addFormValidator.resetForm();
-  addFormValidator.disableSubmitButton();
+  const cardPopupButton = newCardPopup._submitButton;
+  api
+    .addCard(cardData)
+    .then((cardData) => {
+      const newCard = createCard(cardData);
+      section.addItem(newCard, "prepend");
+      newCardPopup.close();
+      addFormValidator.resetForm();
+      addFormValidator.disableSubmitButton();
+      // card creation here
+    })
+    .catch((err) => {
+      cardPopupButton.textContent = "ERROR";
+      console.error(err);
+    })
+    .finally(() => {
+      cardPopupButton.textContent = "Save";
+    });
+  cardPopupButton.textContent = "Saving...";
 });
 newCardPopup.setEventListeners();
 
+const confirmationPopup = new PopupWithSubmit(".popup-confirm");
+
+confirmationPopup.setEventListeners();
+
 /* ------------------------------ Profile Popup ----------------------------- */
 const newProfilePopup = new PopupWithForm(".popup-profile", (formValues) => {
   userInfo.setUserInfo({
```

Comment:
You need to update the `DOM` only in `then` blocks after a successful response from the server, so that when there is an error, it shouldn’t change